### PR TITLE
RefreshTokenModeSpec closes #1081

### LINKED ISSUE #1081: Assess/Fix/Implement JesBackendSpec should "be verified when localizing with Refresh Token"
This is an extension of #1026, that will not fix this specific test.

If mocking is not possible that it may require that an actual refresh token is used?

### PR DESCRIPTION
**Pull request by ruchim**
The only tests remaining in JesBackendSpec were all related to RefreshToken behavior, so just renaming the spec to reflect what it's testing. The tests in this spec check if workflow options are incomplete and that the refresh token is decrypted when retrieving it.

### CODE REVIEW

File: engine/src/test/scala/cromwell/engine/backend/jes/RefreshTokenModeSpec.scala [hunk 1]
```diff
@@ -37,4 +35,12 @@ class JesBackendSpec extends FlatSpec with Matchers with Mockito with BeforeAndA
       } should have message s"Missing parameters in workflow options: refresh_token"
     }
   }
+
+  "refresh token value" should "be decrypted when asserting workflow options" in {
+    EncryptionSpec.assumeAes256Cbc()
+
+    val goodOptions = WorkflowOptions.fromMap(Map("refresh_token" -> mockToken)).get
+    refreshToken.assertWorkflowOptions(goodOptions.toGoogleAuthOptions)
+    goodOptions.toGoogleAuthOptions.get("refresh_token").get shouldBe s"$mockToken"
```
Comment (Horneth) [inline engine/src/test/scala/cromwell/engine/backend/jes/RefreshTokenModeSpec.scala]: could just be `shouldBe mockToken` ?

**Comment by kshakir** [comment]
`cromwell.engine.backend.jes` isn't the right package (I have no clue what the other files are still doing in this folder). Please move this to `cromwell.backend.impl.jes`.

File: engine/src/test/scala/cromwell/engine/backend/jes/RefreshTokenModeSpec.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ import cromwell.util.EncryptionSpec
 import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
 import org.specs2.mock.Mockito
 
-class JesBackendSpec extends FlatSpec with Matchers with Mockito with BeforeAndAfterAll with WorkflowDescriptorBuilder {
+class RefreshTokenModeSpec extends FlatSpec with Matchers with Mockito with BeforeAndAfterAll with WorkflowDescriptorBuilder {
   val testWorkflowManagerSystem = new CromwellTestkitSpec.TestWorkflowManagerSystem()
```
Comment (kshakir) [inline engine/src/test/scala/cromwell/engine/backend/jes/RefreshTokenModeSpec.scala]: 1) We no longer need to spin up a whole workflow manager system if we just need an actor system. There's now [`TestKitSuite`](https://github.com/broadinstitute/cromwell/blob/6af5611c27c678bda2b55950e1a3e37c1c766462/core/src/test/scala/cromwell/core/TestKitSuite.scala) for that.

2) This spec doesn't even use the thread pool spun up inside the WMS / actor system. So remove the `testWorkflowManagerSystem` and related cruft, delete all usage of the `BeforeAndAfterAll` trait, and while there I don't think `Mockito` and `workingDisk` are used either.

**Comment by kshakir** [comment]
Specs look good. Some vestigial bits from reusing old code, but overall 👍

**Comment by Horneth** [comment]
👍

File: engine/src/test/scala/cromwell/engine/backend/jes/RefreshTokenModeSpec.scala (hunk 2 above)
Comment (ruchim) [inline engine/src/test/scala/cromwell/engine/backend/jes/RefreshTokenModeSpec.scala]: @kshakir I didn't realize the TestKitSuite was so lightweight and easy to use, thanks. TIL

**Pull request merged**